pkg/store/config.go

```diff
@@ -4,8 +4,11 @@ import "log"
 
 // StoreConfig represents the configuration of the underlying Store.
 type StoreConfig struct {
-	Dir    string      // The working directory for raft.
-	Tn     Transport   // The underlying Transport for raft.
-	ID     string      // Node ID.
-	Logger *log.Logger // The logger to use to log stuff.
+	Dir          string      // The working directory for raft.
+	Tn           Transport   // The underlying Transport for raft.
+	ID           string      // Node ID.
+	Logger       *log.Logger // The logger to use to log stuff.
+	BasicAuth    bool
```

Comment:
I don't think this is the right way to implement it, we should make an interface called `Auth`, which can provide the check/add/remove permissions, init auth and so on, which looks `struct CredentialsStore`, when you call the `store.Check()`, `auth.Check()` will be called.

Reply:
got it , I'm going to implement a auth registry.